components/cards/SessionCard.tsx

```diff
@@ -0,0 +1,271 @@
+ * @param handleBookMark: function that bookmarks and returns nothing
+ * @param item: object of type T
+ * @param screen: 'home' | 'sessions', default is 'home'
+ * @param variant: 'card' | 'list'
+ * @returns SessionCard Component
+ */
+
+const SessionCardOnHome = (props: Omit<SessionCardHome, 'screen'>) => {
+  const { handlePress, item } = props;
+  const { colors } = useTheme();
+
+  return (
+    <TouchableWithoutFeedback testID="card" onPress={handlePress} disabled={item.slug.length < 1}>
+      <View style={[styles.card, { backgroundColor: colors.card }]}>
+        <Image
+          source={{ uri: item.session_image || '' }}
+          style={styles.image}
+          contentFit="cover"
+          contentPosition="left"
+        />
+        <Space size={8} />
+        <View style={styles.bottom}>
+          <View style={styles.description}>
+            <StyledText font="bold" numberOfLines={2} style={styles.title}>
+              {truncate(50, item.title)}
+            </StyledText>
+          </View>
+
+          <Space size={12} />
+
+          <StyledText size="sm" font="light">
+            {getSessionTimeAndLocation(item.slug)}
+          </StyledText>
+        </View>
+      </View>
+    </TouchableWithoutFeedback>
+  );
+};
+
+const SessionCardOnSessions = (props: Omit<SessionCardSessions, 'screen'>) => {
+  const { handlePress, item, handleBookMark } = props;
+  const { colors } = useTheme();
+  return (
+    <TouchableWithoutFeedback testID="card-sessions" onPress={handlePress} disabled={item.slug.length < 1}>
+      <View style={[styles.card, styles.cardTertiary, { backgroundColor: colors.card }]}>
+        {item.session_image !== null ? (
+          <Image
+            source={{
+              uri:
+                item.session_image ||
+                'https://res.cloudinary.com/droidconke/image/upload/v1668083311/prod/upload/sessions/pwt9pnojtmng8bymxkls.png',
+            }}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="left"
+          />
+        ) : (
+          <Image
+            source={require('../../assets/images/banner.png')}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="center"
+          />
+        )}
+
+        <Space size={4} />
+
+        <View style={styles.bottomForSessions}>
+          <StyledText size="sm" font="light">
+            {getScheduleTimeAndLocation(item.start_date_time, item.end_date_time, item.rooms[0])}
+          </StyledText>
+          <Space size={11} />
+          <View>
+            <StyledText font="bold" numberOfLines={2} style={styles.title}>
+              {truncate(50, item.title)}
+            </StyledText>
+          </View>
+          <Space size={16} />
+          <Row>
+            <Row style={styles.avatarRow}>
+              {item.speakers.map((speaker) => (
+                <Link key={speaker.avatar} href={{ pathname: `/${speaker.name}`, params: { name: speaker.name } }}>
```

Comment:
this is still throwing an error.
if you do it as below 
```
<Link key={speaker.avatar} href={{ pathname: `/${speaker.name}` }}>
```

it redirects to the speaker page but then pushes a session route. I'll look into this later

Reply:
lol so I have faced that error for the past 1 and a half hours, scoured through expo router docs trying to see if it's possible to pass a whole object. I think I have now figured out what you where trying to convey

Reply:
good job, this works